sdk/python/pvsite_datamodel/sqlmodels.py

```diff
@@ -73,20 +77,20 @@ class GenerationSQL(Base, CreatedMixin):
 
     __tablename__ = "generation"
 
-    generation_uuid = sa.Column(UUID(as_uuid=True), default=uuid.uuid4, primary_key=True)
-    site_uuid = sa.Column(
-        UUID(as_uuid=True), sa.ForeignKey("sites.site_uuid"), nullable=False, default=uuid.uuid4
+    generation_uuid = sa.Column(
+        UUID(as_uuid=True), default=uuid.uuid4, primary_key=True
     )
-    power_kw = sa.Column(REAL, nullable=False)
-    datetime_interval_uuid = sa.Column(
+    site_uuid = sa.Column(
         UUID(as_uuid=True),
-        sa.ForeignKey("datetime_intervals.datetime_interval_uuid"),
+        sa.ForeignKey("sites.site_uuid"),
         nullable=False,
-        default=uuid.uuid4,
-    )
-    datetime_interval: DatetimeIntervalSQL = relationship(
-        "DatetimeIntervalSQL", back_populates="generation"
+        index=True,
     )
+    generation_power_kw = sa.Column(sa.Float, nullable=False)
+
+    start_utc = sa.Column(sa.DateTime, nullable=False, index=True)
+    end_utc = sa.Column(sa.DateTime)
```

Comment:
could index on this too? But leave it for the moment

Reply:
Yeah I would wait and see if we do actually need to search on the `end_utc` field. If so, and if we also always filter on `start_utc`, maybe we should consider a multi-column index on (start_utc, end_utc); but I'm not absolutely sure this is actually a right use-cse for this.